packages/core/admin/server/src/services/role.ts

```diff
@@ -376,7 +379,8 @@ const addPermissions = async (roleId: Entity.ID, permissions: any) => {
 
   const permissionsWithRole = permissions
     .map(set('role', roleId))
-    // @ts-expect-error
+    // @ts-expect-error - refactor domain/permission Condition type, as it's now expecting
+    // a string but it should be a Condition interface
```

Comment:
I think at some point we need to discuss what do we do about properties that:
- Are created using a string
- Once created they are returned as a different object

The condition attribute is one example. And I just omitted the work for now as I don't think it's extremely necessary now.

I think in the future we could just create different definitions for the CreatePayload and the actual DB data
@jhoward1994